### PR TITLE
Fix clang-tidy misc-unconventional-assign-operator warnings.

### LINKED ISSUE #22144: Fix misc-unconventional-assign-operator errors.
There are 4 clang-tidy errors caused by the return types of various property assignment operators.
This issue is to fix that.

### PR DESCRIPTION
**Pull request by edwardb96**
Clang tidy issues warnings for unconventional assignment operators which don't return `*this`. This PR fixes a couple of instances where this is the case within Mantid.

**To test:**
1. Ensure builds pass.
2. Code Inspection

Fixes #22144. 

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by martyngigg — COMMENTED** [review]
@LamarMoore @SimonHeybrock Just want to check you're okay with the changes here r.e. `IndexProperty` and `IndexInfo`?

File: Framework/Indexing/src/IndexInfo.cpp [hunk 1]
```diff
@@ -90,7 +90,8 @@ IndexInfo::~IndexInfo() = default;
 
 IndexInfo &IndexInfo::operator=(const IndexInfo &other) {
   auto copy(other);
-  return *this = std::move(copy);
+  *this = std::move(copy);
+  return *this;
```
Comment (SimonHeybrock) [inline Framework/Indexing/src/IndexInfo.cpp]: Isn't the old version doing the same thing?

**Review by SimonHeybrock — COMMENTED** [review]
@martyngigg Regarding the properties, what `IndexProperty` does seems like it was simply copied from the behavior of `PropertyWithValue`, i.e., the question we should ask is why `PropertyWithValue::operator=` was written in this way?

File: Framework/Indexing/src/IndexInfo.cpp (hunk 1 above)
Comment (edwardb96) [reply Framework/Indexing/src/IndexInfo.cpp]: Functionally they are the same yes. Clang Tidy doesn't see that though. This is arguably more readable at a quick glance.

**Comment by martyngigg** [comment]
@SimonHeybrock Good point and not one I can answer as no one is left from when that code was originated. If I'm not mistaken the only situations that should be affected are chained assignments, right? I don't think we do that anywhere and I certainly can't see why we'd actually want the current non-standard behaviour so I'm happy for this to be changed - would you agree?

**Comment by SimonHeybrock** [comment]
I agree. In the case of chained assignment I assume we would either get a compiler failure, or there is an implicit conversion from `PropertyWithValue` back to string that covers such cases? In any case I do not see a reason to keep this, given that it compiles and tests pass.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**